jaeger-core/src/main/java/com/uber/jaeger/Tracer.java

```diff
@@ -529,9 +566,29 @@ public Builder withBaggageRestrictionManager(BaggageRestrictionManager baggageRe
     }
 
     public Tracer build() {
-      return new Tracer(this.serviceName, reporter, sampler, registry, clock, metrics, tags,
+      if (reporter == null) {
+        reporter = new RemoteReporter.Builder()
+            .withMetrics(metrics)
+            .build();
+      }
+      if (sampler == null) {
+        sampler = new RemoteControlledSampler.Builder(serviceName)
+            .withMetrics(metrics)
+            .build();
+      }
+      if (withLoggingSpans) {
```

Comment:
copying from other PR: I prefer not to have this option on the Tracer.Builder, it creates unnecessary coupling with CompositeReporter, LoggingReporter, etc. This feature can be left to Config classes only.

Reply:
I would recommend to people start using `Tracer.Builder` When they use programmatic API. I think it's useful to have it here directly. I am not sure if the composite reporter is easy enough to find/use.